utils/schema/generate.js

```diff
@@ -0,0 +1,160 @@
+function createDefinitions(resources, lookups, additionalProperties = false) {
+  const typeMappings = {
+    'Edm.String': 'string',
+    'Edm.Boolean': 'boolean',
+    'Edm.Int16': 'integer',
+    'Edm.Int32': 'integer',
+    'Edm.Int64': 'integer',
+    'Edm.Decimal': 'number',
+    'Edm.Single': 'number',
+    'Edm.Double': 'number',
+    'Edm.DateTimeOffset': 'string',
+    'Edm.Date': 'string'
+  };
+
+  // Preprocess the lookups data to create a hashmap
+  const lookupsMap = {};
+  lookups.forEach(lookup => {
+    if (!lookupsMap[lookup.lookupName]) {
+      lookupsMap[lookup.lookupName] = new Set();
+    }
+    lookupsMap[lookup.lookupName].add(lookup.lookupValue);
+  });
+
+  function getPossibleValuesForNamespacedType(lookupName) {
+    const possibleValues = lookupsMap[lookupName];
+    return possibleValues ? [...possibleValues] : [];
+  }
+
+  const definitions = {};
+
+  for (const [resourceName, resourceFields] of Object.entries(resources)) {
+    const properties = {};
+
+    resourceFields.forEach(field => {
+      const { fieldName } = field;
+      let schema = {};
+      if (field.isExpansion || field.isCollection) {
+        const itemTypeSchema = {};
+
+        if (field.isExpansion) {
+          itemTypeSchema['$ref'] = `#/definitions/${field.typeName}`;
+        } else {
+          const mappedType = typeMappings[field.type] || 'object';
+          itemTypeSchema['type'] = mappedType;
+          if (field.maxLength) itemTypeSchema['maxLength'] = field.maxLength;
+          if (field.type === 'Edm.DateTimeOffset') itemTypeSchema['format'] = 'date-time';
+          if (field.type === 'Edm.Date') itemTypeSchema['format'] = 'date';
+
+          // probably needs to be another condition?
+          if (field.type.startsWith('org.reso.metadata')) {
```

Comment:
One thing on the namespace - it won't always start with `org.reso.metadata` - this will be dependent on whatever the namespaces they're using are. That being said, this case will only exist when we're working with OData-based metadata and have to resolve the types for things like enumerations or complex types. 

We should probably also turn the `org.reso.metadata` namespace into a constant.